packages/garbo/src/tasks/cockroachPrep.ts

```diff
@@ -16,35 +16,83 @@ import {
 import {
   abort,
   adv1,
+  Effect,
+  effectModifier,
+  isShruggable,
+  Item,
   mallPrice,
+  myAdventures,
   myBuffedstat,
+  myEffects,
+  retrieveItem,
   runChoice,
+  Stat,
+  toEffect,
+  use,
   visitUrl,
 } from "kolmafia";
 import { garboValue } from "../garboValue";
 import { freeFightFamiliar } from "../familiar";
-import { freeFightOutfit } from "../outfit";
+import { freeFightOutfit, meatTargetOutfit } from "../outfit";
 import { GarboStrategy, Macro } from "../combat";
 import { acquire } from "../acquire";
 
+function getBestDebuffItem(stat: Stat): Item {
+  const debuffs = Item.all()
+    .map((item) => ({ item, effect: effectModifier(item, "Effect") }))
+    .filter(
+      ({ effect }) =>
+        effect !== $effect.none &&
+        !have(effect) &&
+        getModifier(stat.toString(), effect) < 0,
+    );
+
+  return maxBy(
+    debuffs,
+    ({ item, effect }) =>
+      mallPrice(item) / getModifier(stat.toString(), effect),
```

Comment:
```
> js Item.all().filter((i) => effectModifier(i, "Effect") !== Effect.none && mallPrice(i) < 0)

Returned: aggregate item [4]
0 => Bee's Knees
1 => Sockdollager
2 => Ish Kabibble
3 => Hot Socks
```
[garf]

Is the easiest fix to this to check for item type potion?

Reply:
We don't want to diet here at all, so a filter for item.potion is a good idea. Updated.